tests/test_providers.py

```diff
@@ -312,3 +324,632 @@ def parse_(self, response: DummyResponse, page: GeoProductPage):  # type: ignore
     item, url, _ = await crawl_single_item(GeoZyteAPISpider, HtmlResource, settings)
     assert item is None
     assert "Geolocation dependencies must be annotated" in caplog.text
+
+
+@defer.inlineCallbacks
+def run_provider(server, to_provide, settings_dict=None, request_meta=None):
+    class AnyResponseSpider(Spider):
+        name = "any_response"
+
+    request = Request(server.urljoin("/some-page"), meta=request_meta)
+    settings = create_scrapy_settings()
+    settings["ZYTE_API_URL"] = server.urljoin("/")
+    if settings_dict:
+        settings.update(settings_dict)
+    crawler = get_crawler(settings, AnyResponseSpider)
+    yield from crawler.engine.open_spider(crawler.spider)
+    injector = Injector(crawler)
+    provider = ZyteApiProvider(injector)
+
+    coro = provider(to_provide, request, crawler)
+    results = yield defer.Deferred.fromFuture(asyncio.ensure_future(coro))
+
+    return results
+
+
+@defer.inlineCallbacks
+def test_provider_any_response(mockserver):
+    # Use only one instance of the mockserver for faster tests.
+    def provide(*args, **kwargs):
+        return run_provider(mockserver, *args, **kwargs)
+
+    results = yield provide(set())
+    assert results == []
+
+    # AnyResponse would use HttpResponse by default, if neither BrowserResponse
+    # nor HttpResponse is available.
+    results = yield provide(
+        {
+            AnyResponse,
+        }
+    )
+    assert len(results) == 1
+    assert type(results[0]) == AnyResponse
+    assert type(results[0].response) == HttpResponse
+
+    # Same case as above, HttpResponse is used by default
+    results = yield provide({AnyResponse, Product})
+    assert len(results) == 2
+    assert type(results[0]) == AnyResponse
+    assert type(results[1]) == Product
```

Comment:
Should there be an assert for `type(results[0].response) == HttpResponse`? Or should it be BrowserResponse here (i.e. the comment above is outdated)?

Reply:
I've removed this test since they duplicate the other test cases that follow it. Not to mention these also lack certain cases. https://github.com/scrapy-plugins/scrapy-zyte-api/pull/161/commits/df32f14598238a0166e8ba488c45d1857e497e55